akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -148,10 +149,7 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
 
   private final class SnapshotFilenameFilter(persistenceId: String) extends FilenameFilter {
     def accept(dir: File, name: String): Boolean = {
-      name match {
-        case FilenamePattern(pid, snr, tms) ⇒ pid.equals(URLEncoder.encode(persistenceId))
-        case _                              ⇒ false
-      }
+      extractMetadata(name).exists(_._1.equals(persistenceId))
```

Comment:
Ad. 1 You mean allowing some false positives through this filter and removing them later if it turns out that `extractMetadata` returns `None`?

Ad. 2 You're right, changed it.